flexbox/src/main/java/com/google/android/flexbox/FlexboxLayoutManager.java

```diff
@@ -368,8 +368,10 @@ public void setAlignContent(@AlignContent int alignContent) {
 
     @Override
     public int getDecorationLengthMainAxis(View view, int index, int indexInFlexLine) {
+        View v = mRecycler.getViewForPosition(index);
+        calculateItemDecorationsForChild(v, TEMP_RECT);
         if (isMainAxisDirectionHorizontal()) {
-            return getLeftDecorationWidth(view) + getRightDecorationWidth(view);
+            return getLeftDecorationWidth(v) + getRightDecorationWidth(v);
```

Comment:
Thank you.I have modify the code.
But i have to say.
view = mRecycler.getViewForPosition(index);
calculateItemDecorationsForChild(view, TEMP_RECT);
Here actually is not the best solution because calculateItemDecorationsForChild is invoked frequent.But it is a solution.
At last. I write the test code and test on HUAWEI and SAMSUNG devices.